collect_app/src/main/java/org/odk/collect/android/configure/qr/QRCodeMenuProvider.kt

```diff
@@ -44,11 +47,20 @@ class QRCodeMenuDelegate internal constructor(
         }
     }
 
-    override fun onCreateOptionsMenu(menuInflater: MenuInflater, menu: Menu) {
+    @SuppressLint("RestrictedApi")
```

Comment:
It's mentioned briefly in the Material Components docs [here](https://github.com/material-components/material-components-android/blob/master/docs/components/Menu.md#adding-icons-to-popup-menus). Looks like this is currently the only way, but that they might add a public API in the future. For the moment, could we pull out our own helper that calls `setOptionalIconsVisible` (probably should live in `material`) so that we only have the risky code (and the `@SupressLint`) in one place?

Reply:
> It's mentioned briefly in the Material Components docs [here](https://github.com/material-components/material-components-android/blob/master/docs/components/Menu.md#adding-icons-to-popup-menus).

Ah right.

>  For the moment, could we pull out our own helper that calls setOptionalIconsVisible (probably should live in material) so that we only have the risky code (and the @SupressLint) in one place?

I've just added an extension function for that. It's placed in the `material` module but I'm not sure maybe it should be `androidshared`?

Reply:
Ah yeah good point. I guess `MenuProvider` is more of "android thing" than a "material thing".

Reply:
Fixed.